usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MedicationManagerClient.java

```diff
@@ -0,0 +1,216 @@
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param Medication The Medication entity that contains the attributes of the Medication of the pet
+     * @return The response code
+     * @throws ExecutionException When the retrieval fails
+     * @throws InterruptedException When the retrieval is interrupted
+     */
+    public int createMedication(String accessToken, String owner, String petName, Medication Medication)
+            throws ExecutionException,
+            InterruptedException {
+        JSONObject reqJson = Medication.getBody().buildMedicationJson();
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(POST);
+        taskManager.setReqBody(reqJson);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + dash + petName + dash
+                        + Medication.getDate() + dash + Medication.getName(), accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Deletes the medication of an specified pet and an specified owner.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the Medication was taken
+     * @param name Name of the medication consumed
+     * @return The response code
+     * @throws ExecutionException When the deletion fails
+     * @throws InterruptedException When the deletion is interrupted
+     */
+    public int deleteByDateName(String accessToken, String owner, String petName, DateTime date, String name)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(DELETE);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + dash + petName + dash + date + dash
+                + name, accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Deletes all the Medications of the specified pet from database.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @return The response code
+     * @throws ExecutionException When the deletion fails
+     * @throws InterruptedException When the deletion is interrupted
+     */
+    public int deleteAllMedications(String accessToken, String owner, String petName)
+            throws ExecutionException,
+            InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(DELETE);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + dash + petName, accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403182)